### PR TITLE
Expand Prettier to eliminate pretty-format-json

### LINKED ISSUE #317: Remove the `pretty-format-json` hook from this and other repositories
## Description

This repository has https://github.com/pre-commit/pre-commit-hooks#pretty-format-json hook while other repositories probably do not. The requirements for linting `json` files thus differs between repositories. This difference causes extra sync requests when the `json` files from this repositories have different format than the `json` files in other repositories (and no other changes).

This PR was failing CI:
https://github.com/WordPress/openverse-frontend/pull/1917
I've fixed linting errors, and then merged them in. After that, a new sync PR was opened with identical changes but different formatting:
https://github.com/WordPress/openverse-frontend/pull/1920

This issue is marked as `high` priority because not fixing it causes a drift between repository configs.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #317 by @obulat 

## Description

This PR removes the `pretty-format-json` hook, and expands the coverage of Prettier to include all files except Markdown `.md` files. This means that JSON files will now be formatted by Prettier which 
- preserves unicode such as emoji
- is more standardised in the JS world

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
Great, thank you!

**Review by AetherUnbound — APPROVED** [review]
You had me sold on emojis alone! 🚀

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -73,9 +69,7 @@ repos:
     rev: v2.5.0
     hooks:
       - id: prettier
-        files: ^js/.*$
-      - id: prettier
-        types: [yaml]
+        exclude: .*\.md$
```
Comment (krysal) [inline .pre-commit-config.yaml:72]: Just out of curiosity, why exclude Markdown files?

Comment (dhruvkb) [reply .pre-commit-config.yaml:72]: Prettier has a very specific Markdown style guide that makes a lot of changes to our Markdown files, which would also ripple into other repos. We can enable it after a discussion about the scope of the changes.

Comment (dhruvkb) [reply .pre-commit-config.yaml:72]: See #333.

**Pull request merged**